x-pack/plugins/security/public/authentication/login/login_page.tsx

```diff
@@ -5,45 +5,81 @@
  */
 
 import React, { Component } from 'react';
-
-import { FormattedMessage } from '@kbn/i18n/react';
-
-import {
-  // @ts-ignore
-  EuiCard,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiIcon,
-  EuiSpacer,
-  EuiText,
-  EuiTitle,
-} from '@elastic/eui';
+import ReactDOM from 'react-dom';
 import classNames from 'classnames';
-import { LoginState } from '../../login_state';
-import { BasicLoginForm } from '../basic_login_form';
-import { DisabledLoginForm } from '../disabled_login_form';
+import { BehaviorSubject } from 'rxjs';
+import { parse } from 'url';
+import { EuiFlexGroup, EuiFlexItem, EuiIcon, EuiSpacer, EuiText, EuiTitle } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { CoreStart, FatalErrorsStart, HttpStart } from 'src/core/public';
+import { LoginLayout } from '../../../common/licensing';
+import { BasicLoginForm, DisabledLoginForm } from './components';
+import { LoginState } from './login_state';
 
 interface Props {
-  http: any;
-  window: any;
-  next: string;
-  infoMessage?: string;
-  loginState: LoginState;
-  isSecureConnection: boolean;
-  requiresSecureConnection: boolean;
+  http: HttpStart;
+  fatalErrors: FatalErrorsStart;
   loginAssistanceMessage: string;
+  requiresSecureConnection: boolean;
+}
+
+interface State {
+  loginState: LoginState | null;
 }
 
-export class LoginPage extends Component<Props, {}> {
+const infoMessageMap = new Map([
+  [
+    'SESSION_EXPIRED',
+    i18n.translate('xpack.security.login.sessionExpiredDescription', {
+      defaultMessage: 'Your session has timed out. Please log in again.',
+    }),
+  ],
+  [
+    'LOGGED_OUT',
+    i18n.translate('xpack.security.login.loggedOutDescription', {
+      defaultMessage: 'You have logged out of Kibana.',
+    }),
+  ],
+]);
+
+export class LoginPage extends Component<Props, State> {
+  state = { loginState: null };
+
+  public async componentDidMount() {
+    const loadingCount$ = new BehaviorSubject(1);
```

Comment:
__note:__ that's what I'm experimenting with to replace dependency on the injected vars - requires a bit of code on client and server, but seems to work fine.